ui/server/src/test/scala/pl/touk/nussknacker/ui/component/DefaultComponentIdProviderSpec.scala

```diff
@@ -0,0 +1,87 @@
+package pl.touk.nussknacker.ui.component
+
+import org.scalatest.{FlatSpec, Matchers}
+import pl.touk.nussknacker.engine.api.component.ComponentType.ComponentType
+import pl.touk.nussknacker.engine.api.component.{ComponentId, ComponentType, SingleComponentConfig}
+import pl.touk.nussknacker.engine.graph.node._
+import pl.touk.nussknacker.engine.graph.service.ServiceRef
+import pl.touk.nussknacker.engine.graph.sink.SinkRef
+import pl.touk.nussknacker.engine.graph.source.SourceRef
+import pl.touk.nussknacker.engine.graph.subprocess.SubprocessRef
+import pl.touk.nussknacker.test.PatientScalaFutures
+
+class DefaultComponentIdProviderSpec extends FlatSpec with Matchers with PatientScalaFutures {
+
+  import org.scalatest.prop.TableDrivenPropertyChecks._
+  import pl.touk.nussknacker.engine.spel.Implicits._
+
+
+  private val processingType = "testProcessingType"
+  private val componentName = "testComponentName"
+  private val componentNameToOverride = "componentNameToOverride"
+  private val overriddenId = ComponentId("overriddenId")
+
+  private val componentIdProvider = new DefaultComponentIdProvider(Map(
+    processingType -> Map(
+      componentNameToOverride -> SingleComponentConfig.zero.copy(componentId = Some(overriddenId))
+    )
+  ))
+
+  private val baseComponentsType = List(
+    ComponentType.Filter, ComponentType.Split, ComponentType.Switch, ComponentType.Variable,
+    ComponentType.MapVariable, ComponentType.FragmentInput, ComponentType.FragmentOutput
+  )
+
+  private val componentsType = List(
+    ComponentType.Processor, ComponentType.Enricher, ComponentType.Sink, ComponentType.Source,
+    ComponentType.Fragments, ComponentType.CustomNode,
+  )
+
+  it should "create ComponentId" in {
+    val testingData = Table(
+      ("componentsType", "name", "expected"),
+      (baseComponentsType, componentName, baseComponentsType.map(cid)),
+      (baseComponentsType, componentNameToOverride, baseComponentsType.map(cid)),
+      (componentsType, componentName, componentsType.map(cid)),
+      (componentsType, componentNameToOverride, componentsType.map(_ => overriddenId)),
+    )
+
+    forAll(testingData) { (componentsType: List[ComponentType], name: String, expected: List[ComponentId]) =>
+      val result = componentsType.map(componentIdProvider.createComponentId(processingType, name, _))
+      result shouldBe expected
+    }
+  }
+
+  it should "create ComponentId for NodeData" in {
```

Comment:
I wonder, if we moved implementation to this class, and have tests added here, shouldn't we remove/simplify some tests in ComponentService?